runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1027,19 +1604,49 @@ TR_VectorAPIExpansion::expandVectorAPI()
       TR::Node *parent = NULL;
       TR::MethodSymbol *methodSymbol = NULL;
 
-      if (opCodeValue == TR::treetop || opCodeValue == TR::NULLCHK)
+      if (opCodeValue == TR::treetop || opCodeValue == TR::NULLCHK ||
+          (boxingAllowed() && treeTopAllowedWithBoxing(opCodeValue)))
           {
           parent = node;
           node = node->getFirstChild();
           opCodeValue = node->getOpCodeValue();
           }
 
+      if (node->chkStoredValueIsIrrelevant())
+         continue;
+
       TR::ILOpCode opCode = node->getOpCode();
 
-      if (opCodeValue != TR::astore && !opCode.isFunctionCall())
+      if (opCode.isFunctionCall() && node->getSymbolReference()->getReferenceNumber() == TR_prepareForOSR) // TODO
          continue;
 
-      if (node->chkStoredValueIsIrrelevant())
+      TR::DataType elementType;
+      int32_t bitsLength;
+      vapiObjType objectType;
+
```

Comment:
Can you limit the scope of these variables or use other names?
Variables with the same names are declared and used in the `if (opCodeValue == TR::astore) { ... }` block below.

Reply:
Done